include/odp/api/spec/packet.h

```diff
@@ -1979,6 +1989,147 @@ int8_t odp_packet_shaper_len_adjust(odp_packet_t pkt);
  */
 void odp_packet_shaper_len_adjust_set(odp_packet_t pkt, int8_t adj);
 
+/*
+ *
+ * Packet vector handling routines
+ * ********************************************************
+ *
+ */
+
+/**
+ * Get packet vector handle from event
+ *
+ * Converts an ODP_EVENT_PACKET_VECTOR type event to a packet vector handle
+ *
+ * @param ev Event handle
+ * @return Packet vector handle
+ *
+ * @see odp_event_type()
+ */
+odp_packet_vect_t odp_packet_vect_from_event(odp_event_t ev);
+
+/**
+ * Convert packet vector handle to event
+ *
+ * @param vpkt Packet vector handle
+ *
+ * @return Event handle
+ */
+odp_event_t odp_packet_vect_to_event(odp_packet_vect_t vpkt);
+
+/**
+ * Allocate a packet vector from a packet vector pool
+ *
+ * Allocates a packet vector from the specified packet vector pool.
+ * The pool must have been created with the ODP_POOL_PACKET_VECTOR type.
+ *
+ * @param pool Packet vector pool handle
+ *
+ * @return Handle of allocated packet vector
+ * @retval ODP_PACKET_VECT_INVALID  Packet vector could not be allocated
+ */
+odp_packet_vect_t odp_packet_vect_alloc(odp_pool_t pool);
+
+/**
+ * Free packet vector
+ *
+ * Frees the packet vector into the packet vector pool it was allocated from.
+ *
+ * @param vpkt Packet vector handle
+ */
+void odp_packet_vect_free(odp_packet_vect_t vpkt);
```

Comment:
Need to document if packets in the vector are freed or not. I think this basic free call would just free the vector event (ignores all packet handles). We could then have another free call that frees all packets and the vector.

Reply:
How about adding an argument for this function instead of a new function. say bool free_elements or so?

Reply:
odp_event_free() should work correctly for any event, so it should free all packets inside a vector.

Reply:
odp_packet_vector_free() would be the basic form that frees only the container. odp_packet_vector_free_all() ??? could be added later for freeing both container and all remaining packet.